core/src/apps/cardano/native_script.py

```
@@ -1,105 +1,111 @@
         else:
             raise INVALID_NATIVE_SCRIPT
-    elif script.type == CardanoNativeScriptType.ALL:
-        for sub_script in script.scripts:
+    elif script_type == CNST.ALL:
+        for sub_script in scripts:
             validate_native_script(sub_script)
-    elif script.type == CardanoNativeScriptType.ANY:
-        for sub_script in script.scripts:
+    elif script_type == CNST.ANY:
+        for sub_script in scripts:
             validate_native_script(sub_script)
```

Ruff (0.8.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)